spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/TopicStep.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+/**
+ * Exception thrown when a test times out.
+ *
+ * @author Jonas Geiregat
+ */
+public interface TopicStep {
```

Comment:
Since the type `T` is required in all the places, we can simply mark these spec interfaces (the other ones as well) w/ that type `(ex. TopicStep<T>`) and that way there will be no need to cast down in the other steps.

Reply:
See https://github.com/spring-projects/spring-pulsar/pull/582#discussion_r1501707626